udata/features/territories/views.py

```diff
@@ -26,96 +24,84 @@ def dict_to_namedtuple(name, data):
-
-    csvfile_out = StringIO.StringIO()
-    with open(resource_path, 'rb') as csvfile_in:
-        reader = csv.DictReader(csvfile_in, delimiter=str(';'))
-        writer = csv.DictWriter(csvfile_out, fieldnames=reader.fieldnames)
-        writer.writerow(dict(zip(writer.fieldnames, writer.fieldnames)))
-        for row in reader:
-            csv_column = row[args['csv_column']].encode('utf-8')
-            # `lstrip` ensure comparison for counties.
-            if csv_column.lstrip('0') == match.lstrip('0'):
-                writer.writerow(row)
-
-    csvfile_out.seek(0)  # Back to 0 otherwise the file is served empty.
-    attachment_filename = '{territory_name}_{resource_name}.csv'.format(
-        territory_name=territory.name,
-        resource_name=resource.title.replace(' ', '_'))
-    return send_file(csvfile_out, as_attachment=True,
-                     attachment_filename=attachment_filename)
-
-
-@blueprint.route('/territory/regions/', endpoint='home')
+@blueprint.route('/territoires/', endpoint='home')
 def render_home():
     if not current_app.config.get('ACTIVATE_TERRITORIES'):
         return abort(404)
 
-    regions = GeoZone.objects.get(id='country/fr').children
+    regions = GeoZone.objects(level='fr/region').valid_at(date.today())
+    regions = sorted(
+        regions,
+        key=lambda zone: unicodedata.normalize('NFD', zone.name)
+                                    .encode('ascii', 'ignore'))
 
     return theme.render('territories/home.html', **{
         'geojson': {
             'type': 'FeatureCollection',
             'features': [region.toGeoJSON() for region in regions]
-        }
+        },
+        'regions': regions
     })
 
 
-@blueprint.route('/town/<int:code>/', endpoint='town')
+@blueprint.route('/town/<code>/', endpoint='town')
 def redirect_town(code):
     """
-    Legacy redirect now prefixed with `territories`.
+    Legacy redirect now prefixed with `territories` + French name.
     """
     # Turn the dict into a namedtuple to be consistent in routing.
-    territory = dict_to_namedtuple('Territory',
-                                   {'code': code, 'level_name': 'town'})
+    territory = dict_to_namedtuple(
+        'Territory', {'code': str(code), 'level_name': 'commune'})
+    return redirect(url_for('territories.territory', territory=territory))
+
+
+@blueprint.route('/territoires/<level>/<code>@latest/',
+                 endpoint='redirect_territory')
+def redirect_territory(level, code):
+    """
+    Implicit redirect given the INSEE code.
+
+    Optimistically redirect to the latest valid/known INSEE code.
+    """
+    territory = GeoZone.objects.valid_at(datetime.now()).filter(
+        code=code, level='fr/{level}'.format(level=level)).first()
     return redirect(url_for('territories.territory', territory=territory))
 
 
-@blueprint.route('/territory/<territory:territory>/', endpoint='territory')
+@blueprint.route('/territoires/<territory:territory>/', endpoint='territory')
```

Comment:
No, the discussion was on the level names and it was the initial purpose of this pull-request.
What was discussed and agreed is to have french level names because there is no strict equivalent translation. The prefix change was not discussed and decided.
More, this is the only translated URL here: still using `town`(which, as opposite, because it's level name, should be translated)